frame/nomination-pools/src/lib.rs

```diff
@@ -1571,8 +1593,23 @@ pub mod pallet {
 			T::WeightInfo::bond_extra_transfer()
 			.max(T::WeightInfo::bond_extra_reward())
 		)]
-		pub fn bond_extra(origin: OriginFor<T>, extra: BondExtra<BalanceOf<T>>) -> DispatchResult {
+		pub fn bond_extra(
+			origin: OriginFor<T>,
+			other: Option<AccountIdLookupOf<T>>,
+			extra: BondExtra<BalanceOf<T>>,
+		) -> DispatchResult {
```

Comment:
Sorry, I meant to suggest reuse the internal code. Otherwise we want to have two different transactions to prevent this being a breaking change.

The new one should be exactly what you have here, and support both self-claim and other-claim. 

The old one should remain as-is, but the docs should encourage using the new one. 

these two should internally use the same helper function, using probably `other: Option<AccountIdLookupOf<T>>,` being `Some()` in one and `None` the other.